pyteal/ast/abi/reference_type.py

```diff
@@ -0,0 +1,119 @@
+from abc import abstractmethod
+from typing import cast, Any
+
+from pyteal.errors import TealInputError
+
+from pyteal.ast.expr import Expr
+from pyteal.ast.txn import Txn
+
+from pyteal.ast.abi.uint import Uint, UintTypeSpec
+
+
+class ReferenceTypeSpec(UintTypeSpec):
+    def __init__(self) -> None:
+        return super().__init__(8)
+
+
+ReferenceTypeSpec.__module__ = "pyteal"
```

Comment:
nit: some ignorable comments
- can be inherited from `uint8`
- `__str__` and `new_instance` not yet defined, probably defined as abstract method?

Reply:
can it be? I just tried and maybe confused myself but if I use uint8 I have no opportunity to set the typespec. I _think_ I need to follow the model of abi.Byte